sdk/sdk/src/main/java/org/apache/carbondata/sdk/file/CSVCarbonWriter.java

```diff
@@ -72,6 +93,36 @@ public void write(Object object) throws IOException {
     }
   }
 
+  /**
+   * Load data of all or selected csv files at given location iteratively.
+   *
+   * @throws IOException
+   */
+  @Override
+  public void write() throws IOException {
+    if (this.dataFiles == null || this.dataFiles.length == 0) {
+      throw new RuntimeException("'withCsvPath()' must be called to support load files");
+    }
+    this.csvParser = SDKUtil.buildCsvParser(this.configuration);
+    Arrays.sort(this.dataFiles, Comparator.comparing(CarbonFile::getPath));
+    for (CarbonFile dataFile : this.dataFiles) {
+      this.loadSingleFile(dataFile);
+    }
+  }
+
+  private void loadSingleFile(CarbonFile file) throws IOException {
+    this.csvParser.beginParsing(FileFactory.getDataInputStream(file.getPath(), -1, configuration));
```

Comment:
InputStream is not closed. same as above.

Reply:
closed.